src/Core/BaseEntityModel/EntityReflexion.php

```diff
@@ -2,28 +2,79 @@
 
 namespace Wepesi\Core\BaseEntityModel;
 
-use Exception;
-use ReflectionClass;
+use Wepesi\Core\BaseEntityModel\Provider\Contract\EntityInterface;
 
 /**
  *
  */
 trait EntityReflexion
 {
     /**
-     * @param Entity $entity
+     * @param EntityInterface $entity
      * @return array
      */
-    protected function getEntityName(Entity $entity): array
+    private function getEntityName(EntityInterface $entity): array
     {
         try {
-            $class = new ReflectionClass($entity);
-            $object = $class->newInstance();
-            $method = $class->getMethod('getTableName');
+            $reflexion = new \ReflectionClass($entity);
+            $object = $reflexion->newInstance();
+            $method = $reflexion->getMethod('getTableName');
+            $class_name = strtolower($reflexion->getShortName());
             $method->setAccessible(true);
-            return ['table' => $method->invoke($object)];
-        } catch (Exception $ex) {
+            return ['table' => $method->invoke($object),'class' => $class_name];
+        } catch (\Exception $ex) {
             return ['exception' => $ex->getMessage()];
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * @param string $entity_array
+     * @return array|mixed
+     */
+    private function getEntityRelationLink(array $entity_array)
+    {
+        try {
+            $reflexion = new \ReflectionClass($this);
+            $object = $reflexion->newInstance();
+            $table_name = $entity_array['table'];
+            $class_name = $entity_array['class'];
+            if ($reflexion->hasMethod($table_name)){
+                $method = $reflexion->getMethod($table_name);
+                $method->setAccessible(true);
+            } else if ($reflexion->hasMethod($class_name)){
+                $method = $reflexion->getMethod($class_name);
+                $method->setAccessible(true);
+            }else{
+                throw new \Exception('You should implement a relation method for the ' . $class_name);
+            }
+            return $method->invoke($object);
+        } catch (\Exception $ex) {
+            return ['exception' => $ex->getMessage() . ' from class ' . $this->getClassDefinition($this)->class];
+        }
+    }
+
+    /**
+     * @param EntityInterface $entity
+     * @return object|array
+     */
+    private function getClassDefinition($entity): object
```

Comment:
refactor `getEntityRelationLink` and `getClassDefinition` to have only one method